### PR TITLE
[Quant][FX] Lower QConvAddReLU2d for onednn backend

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91155
* #91154
* #91153
* #91152

**Summary**
Add quantization mappings for QConvAddReLU2d for int8 inference for onednn backend. The fusion and lowering is supported only in FX mode.

**Test plan**
```
python -m pytest test_quantization.py -k test_fuse_conv_bn_add_relu_onednn
python -m pytest test_quantization.py -k test_fuse_conv_bn_add_relu_by_default
python -m pytest test_quantization.py -k test_fuse_conv_bn_add_relu_lowering
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: torch/ao/quantization/backend_config/onednn.py [hunk 1]
```diff
@@ -178,6 +178,85 @@ def conv_add_extra_inputs_getter(pattern):
...
+    are assumed to be copied over from root node to the fused node
+    """
+    relu, add_pattern = pattern
+    _, conv, extra_input = add_pattern
+    return [extra_input]
+
+for add_op in [torch.add, operator.add]:
+    conv_configs.append(
+        BackendPatternConfig()
+            ._set_pattern_complex_format((nn.ReLU, (add_op, nn.Conv2d, MatchAllNode)))
```
Comment (jgong5) [inline torch/ao/quantization/backend_config/onednn.py:207]: Always on the right?

Comment (leslie-fang-intel) [reply torch/ao/quantization/backend_config/onednn.py:207]: No, I will add it later. It's still WIP.

**Review by jgong5 — APPROVED** [review]
LGTM

**Comment by leslie-fang-intel** [comment]
@jerryzh168 Could you help to take a look of this PR again?

**Comment by leslie-fang-intel** [comment]
Hi @jerryzh168 Is there any other comments to this PR? Could you help to take a look again?

**Comment by leslie-fang-intel** [comment]
Hi @jerryzh168, Could you also take a review of this PR again?

**Review by jerryzh168 — APPROVED** [review]
LGTM!

**Pull request closed without merging**